pulp_smash/tests/python/api_v2/test_sync_publish.py

```diff
@@ -46,80 +90,92 @@ def verify_sync(self, cfg, call_report):
+        """
+        if selectors.bug_is_untestable(135, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/135')
+        client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['importer_config'] = {
             'feed': constants.PYTHON_PYPI_FEED_URL,
             'package_names': 'shelf-reader',
-        })
-        call_report = utils.sync_repo(cfg, repo_href)
-        with self.subTest():
-            self.verify_sync(cfg, call_report)
-        with self.subTest():
-            self.verify_package_types(cfg, repo_href)
+        }
+        body['distributors'] = [gen_distributor()]
+        repo = client.post(constants.REPOSITORY_PATH, body)
+        self.repos.append(repo)
+        call_report = utils.sync_repo(self.cfg, repo['_href'])
+        with self.subTest(comment='verify the sync succeeded'):
+            self.verify_sync(self.cfg, call_report)
+        with self.subTest(comment='verify content units are present'):
+            self.verify_package_types(self.cfg, repo)
+        repo = get_details(self.cfg, repo)
+        utils.publish_repo(self.cfg, repo)
+
+
+class UploadTestCase(BaseTestCase):
+    """Test whether content can be uploaded to a Python repository."""
+
+    def test_01_first_repo(self):
+        """Create, upload content into and publish a Python repository.
+
+        See:
+
+        * `Pulp #136 <https://pulp.plan.io/issues/136>`_
+        * `Pulp #2334 <https://pulp.plan.io/issues/2334>`_
+        * `Pulp Smash #492 <https://github.com/PulpQE/pulp-smash/issues/492>`_
+        """
+        if selectors.bug_is_untestable(136, self.cfg.version):
+            self.skipTest('https://pulp.plan.io/issues/135')
+        client = api.Client(self.cfg, api.json_handler)
+        body = gen_repo()
+        body['distributors'] = [gen_distributor()]
+        repo = client.post(constants.REPOSITORY_PATH, body)
+        self.repos.append(repo)
+        for url in constants.PYTHON_EGG_URL, constants.PYTHON_WHEEL_URL:
+            unit = utils.http_get(url)
+            import_params = {
+                'unit_key': {'filename': basename(urlparse(url).path)},
+                'unit_type_id': 'python_package',
+            }
+            upload_import_unit(self.cfg, unit, import_params, repo)
+        with self.subTest(comment='verify content units are present'):
+            self.verify_package_types(self.cfg, repo)
+        repo = get_details(self.cfg, repo)
+        utils.publish_repo(self.cfg, repo)
+
+
+def get_details(cfg, repo):
+    """Return detailed information about a Python repository."""
+    return api.Client(cfg).get(repo['_href'], params={
+        'distributors': True,
+        'importers': True,
+    }).json()
+
+
+def get_repo_path(repo):
+    """Return the root path to a published Python repository."""
+    path = urljoin('/pulp/python/web/', repo['id'])
+    path += '/'
+    return urljoin(path, 'simple/')
```

Comment:
It turns out that dropping `simple/` is the correct thing to do.

See: https://pulp.plan.io/issues/140#note-18